ggircs-app/app/components/XmlDiff/ReportSelector.tsx

```diff
@@ -1,50 +1,47 @@
-import { useState, useCallback } from "react";
+import { useState } from "react";
 import { RelayReportObject } from "types";
 import Input from "@button-inc/bcgov-theme/Input";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faCheck, faTimes } from "@fortawesome/free-solid-svg-icons";
 import debounce from "lodash.debounce";
-import { useRouter } from "next/router";
+import { NextRouter } from "next/router";
 
 interface Props {
   diffSide: String;
   allReports: readonly RelayReportObject[];
+  router: NextRouter;
 }
 
 export const ReportSelector: React.FunctionComponent<Props> = ({
   diffSide,
   allReports,
+  router,
 }) => {
-  const router = useRouter();
-
   const [swrsReportIdIsValid, setSwrsReportIdIsvalid] = useState<boolean>(null);
   const [swrsReportId, setSwrsReportId] = useState<number>(
     Number(router.query[`${diffSide}SideId`])
   );
 
-  const handleRouter = useCallback(
-    (id: number, relayId: string, valid: boolean) => {
-      let query;
-      if (valid) {
-        query = {
-          ...router.query,
-          [`${diffSide}SideId`]: id,
-          [`${diffSide}SideRelayId`]: relayId,
-        };
-      } else {
-        query = router.query;
-        delete query[`${diffSide}SideId`];
-        delete query[`${diffSide}SideRelayId`];
-      }
-      if (!relayId) delete query[`${diffSide}SideRelayId`];
-      const url = {
-        pathname: router.pathname,
-        query,
+  const handleRouter = (id: number, relayId: string, valid: boolean) => {
+    let query;
+    if (valid) {
+      query = {
+        ...router.query,
+        [`${diffSide}SideId`]: id,
+        [`${diffSide}SideRelayId`]: relayId,
       };
-      router.push(url, url, { shallow: true });
-    },
-    [router, diffSide]
-  );
+    } else {
+      query = router.query;
+      delete query[`${diffSide}SideId`];
+      delete query[`${diffSide}SideRelayId`];
+    }
+    if (!relayId) delete query[`${diffSide}SideRelayId`];
+    const url = {
+      pathname: router.pathname,
+      query,
+    };
```

Comment:
you're mutating `router.query` here, that's not good. This might be what was causing the bug, and why you had to remove `useCallback` (since this is breaking the way react works).

`query = router.query;` should be `query = {...router.query}`

Reply:
Oooh good eye! I was wondering the whole time why the `useCallback` was causing issues.